client/rest-high-level/src/test/java/org/elasticsearch/client/dataframe/GetDataFrameTransformStatsResponseTests.java

```diff
@@ -0,0 +1,99 @@
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client.dataframe;
+
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.action.TaskOperationFailure;
+import org.elasticsearch.client.dataframe.transforms.DataFrameTransformStateAndStats;
+import org.elasticsearch.client.dataframe.transforms.DataFrameTransformStateAndStatsTests;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.test.ESTestCase;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.elasticsearch.test.AbstractXContentTestCase.xContentTester;
+
+public class GetDataFrameTransformStatsResponseTests extends ESTestCase {
+
+    public void testXContentParser() throws IOException {
+        xContentTester(this::createParser,
+                GetDataFrameTransformStatsResponseTests::createTestInstance,
+                GetDataFrameTransformStatsResponseTests::toXContent,
+                GetDataFrameTransformStatsResponse::fromXContent)
+                .assertEqualsConsumer(GetDataFrameTransformStatsResponseTests::assertEqualInstances)
+                .assertToXContentEquivalence(false)
+                .supportsUnknownFields(true)
+                .randomFieldsExcludeFilter(path -> path.isEmpty() == false)
+                .test();
+    }
+
+    private static GetDataFrameTransformStatsResponse createTestInstance() {
+        int count = randomIntBetween(1, 3);
+        List<DataFrameTransformStateAndStats> stats = new ArrayList<>();
+        for (int i=0; i<count; i++) {
+            stats.add(DataFrameTransformStateAndStatsTests.randomInstance());
+        }
+
+        List<TaskOperationFailure> taskFailures = null;
+        if (randomBoolean()) {
+            taskFailures = new ArrayList<>();
+            int numTaskFailures = randomIntBetween(1, 4);
+            for (int i=0; i<numTaskFailures; i++) {
+                taskFailures.add(new TaskOperationFailure(randomAlphaOfLength(4), randomNonNegativeLong(), new IllegalStateException()));
+            }
+        }
+        List<ElasticsearchException> nodeFailures = null;
+        if (randomBoolean()) {
+            nodeFailures = new ArrayList<>();
+            int numNodeFailures = randomIntBetween(1, 4);
+            for (int i=0; i<numNodeFailures; i++) {
+                nodeFailures.add(new ElasticsearchException("GetDataFrameTransformStatsResponseTests"));
+            }
+        }
+
+        return new GetDataFrameTransformStatsResponse(stats, taskFailures, nodeFailures);
+    }
+
+    private static void toXContent(GetDataFrameTransformStatsResponse response, XContentBuilder builder) throws IOException {
```

Comment:
This (and the other test class implementations like it in this PR) are breaking from the pattern of `implements ToXContentObject` on the class and then make tests around `AbstractXContentTestCase`. 

@davidkyle is there a technical reason for breaking from this pattern?

Reply:
This pattern is used elsewhere is the HLRC code. I prefer it because it because `toXContent` method is only used for testing and doesn't need to be in the actual `GetDataFrameTransformStatsResponse` class and I've taken a hard line view that code shouldn't be in the production class unless it is used there. 

Either way there is a maintenance problem with keeping the client `to/FromXContent` methods in sync with the server code only the integ tests can do that.

Reply:
:+1: